### PR TITLE
Cut 0.0.111

### PR DESCRIPTION
**Pull request by TheBlueMatt**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,68 @@
+# 0.0.111 - 2022-09-XXX - "Saturated with Messages"
+
+## API Updates
+ * Support for relaying onion messages has been added via a new
+   `OnionMessenger` struct when passed as the `OnionMessageHandler` to a
+   `PeerManager`. Pre-encoded onion messages can also be sent and received
+   (#1503, #1640, #1650, #1652, #1688).
```
Comment (valentinewallace) [inline CHANGELOG.md:7]: 1640 is unrelated IIUC

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,68 @@
...
+   output via a new `Balance::CounterpartyRevokedOutputClaimable` (#1495).
+ * Inbound HTLCs for which we do not (yet) have a preimage are now included in
+   `get_claimable_balance` via a `Balance::MaybePreimageClaimableHTLC` (#1673).
+ * Probes which fail prior to being sent over their first hop are correctly
+   failed with a `Event::ProbeFailed` rather than a `PaymentPathFailed` (#1704).
+ * Pending `Event::HTLCHandlingFailed`s are no longer lost on restart (#1700).
+ * HTLCs which fail prior to being sent over their first hop are now marked as
+   retryable via `!PaymentPathFailed::payment_failed_permanently` (#1702).
+ * Dust HTLCs are now considered failed in the payment tracking logic after the
+   commitment transaction confirms, allowing retry on restart (#1691).
```
Comment (valentinewallace) [inline CHANGELOG.md:47]: Are ChannelMonitors safe to remove now?

**Comment by valentinewallace** [comment]
Wondering how people check if any PRs are missing? Maybe we should start tagging all PRs with the release milestone on merge if they aren't tagged already

**Comment by valentinewallace** [comment]
> git log --merges v0.0.110..HEAD

Super handy, thanks!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,69 @@
...
+ * Support for relaying onion messages has been added via a new
+   `OnionMessenger` struct when passed as the `OnionMessageHandler` to a
+   `PeerManager`. Pre-encoded onion messages can also be sent and received
+   (#1503, #1640, #1650, #1652, #1688).
+ * Rate-limiting of outbound gossip syncs has been rewritten to utilize less
+   buffering inside LDK. The new rate-limiting is also used for onion messages
+   to avoid delaying other messages (#1604. #1660, #1683).
+ * Rather than spawning a full OS thread, `lightning-background-processor` has
+   a new `process_events_async` method which takes the place of a
+   `BackgroundProcessor` for those using Rust async (#1657).
```
Comment (tnull) [inline CHANGELOG.md:13]:
```suggestion
   `BackgroundProcessor` for those using Rust's `async` (#1657).
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,69 @@
...
+ * The `Filter::register_output` return value has been removed, as it was
+   very difficult to correctly implemented (i.e. without blocking) (#1663).
```
Comment (tnull) [inline CHANGELOG.md:15]:
```suggestion
   very difficult to correctly implement (i.e., without blocking) (#1663).
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,69 @@
+# 0.0.111 - 2022-09-XXX - "Saturated with Messages"
+
+## API Updates
```
Comment (valentinewallace) [inline CHANGELOG.md:3]: Could mention 1682, log levels are exposed now

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,69 @@
...
+   various `*MessageHandler` traits, rather than hard-coded (#1701, XXX OMs).
+ * `Event::PaymentPathFailed::rejected_by_dest` has been renamed
+   `payment_failed_permanently` (#1702).
+ * `Invoice` now derives the std `Hash` trait (#1575).
+ * `chain::AccessError` now derives the std `Debug` trait (#1709).
+ * `ReadOnlyNetworkGraph::list_{channels,nodes}` have been added largely for
+   users of downstream bindings (#1651).
+ * `ChannelMonitor::get_counterparty_node_id` is now available (#1635).
+
+## Bug Fixes
```
Comment (valentinewallace) [inline CHANGELOG.md:32]: Could note 1659, guess it's very unlikely though

**Review by valentinewallace — COMMENTED** [review]
Lgtm after outstanding feedback

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,68 @@
...
+   `PeerManager`. Pre-encoded onion messages can also be sent and received
+   (#1503, #1640, #1650, #1652, #1688).
+ * Rate-limiting of outbound gossip syncs has been rewritten to utilize less
+   buffering inside LDK. The new rate-limiting is also used for onion messages
+   to avoid delaying other messages (#1604. #1660, #1683).
+ * Rather than spawning a full OS thread, `lightning-background-processor` has
+   a new `process_events_async` method which takes the place of a
+   `BackgroundProcessor` for those using Rust async (#1657).
+ * The `Filter::register_output` return value has been removed, as it was
+   very difficult to correctly implemented (i.e. without blocking) (#1663).
```
Comment (jkczyz) [inline CHANGELOG.md:15]: Probably should mention that you must manually rescan now.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,68 @@
...
+ * `ChannelHandshakeConfig::their_channel_reserve_proportional_millionths` has
+   been added to allow configuring counterparty reserve values (#1619).
+ * The `rust-bitcoin` dependency has been updated 0.29 (#1658).
+ * The `bitcoin_hashes` dependency has been updated 0.11 (#1677).
+ * `ChannelManager::broadcast_node_announcement` has been moved to
+   `PeerManager` (#1699).
+ * `{Init,Node}Features` sent to peers/broadcasted are now fetched via the
+   various `*MessageHandler` traits, rather than hard-coded (#1701, XXX OMs).
```
Comment (jkczyz) [inline CHANGELOG.md:23]: s/XXX/1688

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,68 @@
...
+## Bug Fixes
+ * The script compared with that returned from `chain::Access` was incorrect
+   ~half of the time, causing spurious gossip rejection (#1666).
+ * Pending in-flight HTLCs are now considered when calculating new routes,
+   ensuring, e.g. MPP retries do not take known-saturated paths (#1643).
+ * Counterparty-revoked outputs are now included in `get_claimable_balance`
+   output via a new `Balance::CounterpartyRevokedOutputClaimable` (#1495).
+ * Inbound HTLCs for which we do not (yet) have a preimage are now included in
+   `get_claimable_balance` via a `Balance::MaybePreimageClaimableHTLC` (#1673).
+ * Probes which fail prior to being sent over their first hop are correctly
```
Comment (jkczyz) [inline CHANGELOG.md:41]: s/which/that

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,68 @@
...
+   failed with a `Event::ProbeFailed` rather than a `PaymentPathFailed` (#1704).
+ * Pending `Event::HTLCHandlingFailed`s are no longer lost on restart (#1700).
+ * HTLCs which fail prior to being sent over their first hop are now marked as
```
Comment (jkczyz) [inline CHANGELOG.md:44]: s/which/that

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,68 @@
...
+   retryable via `!PaymentPathFailed::payment_failed_permanently` (#1702).
+ * Dust HTLCs are now considered failed in the payment tracking logic after the
+   commitment transaction confirms, allowing retry on restart (#1691).
+ * On machines with buggy "monotonic" clocks, LDK will no longer panic if time
+   goes backwards (#1692).
+
+## Backwards Compatibility
+ * The new `current_time` argument to `PeerManager` constructors must be set to
+   a UNIX timestamp for upgraded nodes. Only new nodes may use a counter (#1699)
```
Comment (jkczyz) [inline CHANGELOG.md:53]: nit: end with period

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,68 @@
...
+
+## Security
+0.0.111 fixes a denial-of-service vulnerability which is reachable from
+untrusted input in deployments accepting 0conf channels, or via a race-condition
+in deployments creating outbound 0conf channels.
```
Comment (jkczyz) [inline CHANGELOG.md:59]: Should this be a bullet / have a PR number?

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,69 @@
...
+ * The new `current_time` argument to `PeerManager` constructors must be set to
+   a UNIX timestamp for upgraded nodes. Only new nodes may use a counter (#1699)
+XXX: fill this out
+
+## Security
+0.0.111 fixes a denial-of-service vulnerability which is reachable from
+untrusted input in deployments accepting 0conf channels, or via a race-condition
+in deployments creating outbound 0conf channels.
+
+ * LDK versions prior to 0.0.111 may spuriously panic when receiving a block if
```
Comment (ariard) [inline CHANGELOG.md:61]: I think it was introduced with #1401, so should only affect from 0.0.107 to 0.0.111 ?

File: CHANGELOG.md (hunk 5 above)
Comment (valentinewallace) [reply CHANGELOG.md:3]: That's fair, thought the sample used them too but looks like no. Resolving

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,69 @@
...
+   `get_claimable_balance` via a `Balance::MaybePreimageClaimableHTLC` (#1673).
+ * Probes which fail prior to being sent over their first hop are correctly
+   failed with a `Event::ProbeFailed` rather than a `PaymentPathFailed` (#1704).
+ * Pending `Event::HTLCHandlingFailed`s are no longer lost on restart (#1700).
+ * HTLCs which fail prior to being sent over their first hop are now marked as
+   retryable via `!PaymentPathFailed::payment_failed_permanently` (#1702).
+ * Dust HTLCs are now considered failed in the payment tracking logic after the
+   commitment transaction confirms, allowing retry on restart (#1691).
+ * On machines with buggy "monotonic" clocks, LDK will no longer panic if time
+   goes backwards (#1692).
```
Comment (ariard) [inline CHANGELOG.md:49]: I don't know if #1659 should be added or to minor.

**Review by ariard — COMMENTED** [review]
List of PRs SGTM

File: CHANGELOG.md (hunk 13 above)
Comment (ariard) [reply CHANGELOG.md:61]: From a user viewpoint with low knowledge of LDK they might not know under the wood if they're using 0conf or not by default. Nice for them to be able to to establish if they're affected or not with optimal informations from release notes :)

**Review by jkczyz — DISMISSED** [review]

**Review by ariard — APPROVED** [review]

**Review by valentinewallace — DISMISSED** [review]

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,97 @@
+# 0.0.111 - 2022-09-09 - "Saturated with Messages"
```
Comment (jkczyz) [inline CHANGELOG.md:1]: Update date.

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,97 @@
...
+   buffering inside LDK. The new rate-limiting is also used for onion messages
+   to avoid delaying other messages (#1604. #1660, #1683).
+ * Rather than spawning a full OS thread, `lightning-background-processor` has
+   a new `process_events_async` method which takes the place of a
+   `BackgroundProcessor` for those using Rust's async (#1657).
+ * `ChannelManager::get_persistable_update_future` has been added to block on
+   a ChannelManager needing re-persistence in a Rust async environment (#1657).
+ * The `Filter::register_output` return value has been removed, as it was
+   very difficult to correctly implement (i.e., without blocking). Users
+   previously using it should instead pass dependent transactions in via new
```
Comment (jkczyz) [inline CHANGELOG.md:18]: s/new/additional

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,97 @@
...
+ * HTLCs that fail prior to being sent over their first hop are now marked as
+   retryable via `!PaymentPathFailed::payment_failed_permanently` (#1702).
+ * Dust HTLCs are now considered failed in the payment tracking logic after the
+   commitment transaction confirms, allowing retry on restart (#1691).
+ * On machines with buggy "monotonic" clocks, LDK will no longer panic if time
+   goes backwards (#1692).
+
+## Backwards Compatibility
+ * The new `current_time` argument to `PeerManager` constructors must be set to
+   a UNIX timestamp for upgraded nodes, new nodes may use a counter (#1699).
```
Comment (jkczyz) [inline CHANGELOG.md:61]: s/,/;

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,3 +1,97 @@
...
+ * `Balance::CounterpartyRevokedOutputClaimable` will never be generated for
+   channels which were observed to go on-chain with LDK versions prior to
```
Comment (jkczyz) [inline CHANGELOG.md:63]: s/which/that

**Review by jkczyz — DISMISSED** [review]

**Review by jkczyz — APPROVED** [review]

**Review by valentinewallace — APPROVED** [review]

**Pull request merged**